### PR TITLE
esdoc(SourceMapDevToolPlugin): add some esdoc definitions

### PR DESCRIPTION
**Pull request by vankop**
https://github.com/webpack/webpack/issues/3277

**What kind of change does this PR introduce?**

documentation

**Did you add tests for your changes?**

No

**Does this PR introduce a breaking change?**

No

**What needs to be documented once your changes are merged?**

Nothing

### CODE REVIEW

File: lib/SourceMapDevToolPlugin.js [hunk 1]
```diff
@@ -102,12 +154,22 @@ class SourceMapDevToolPlugin {
 			new SourceMapDevToolModuleOptionsPlugin(options).apply(compilation);
 
 			compilation.hooks.afterOptimizeChunkAssets.tap(
-				{
-					name: "SourceMapDevToolPlugin",
-					context: true
-				},
+				//TODO need to fix type of Tap, currently { name: string; type: TapType; fn: Function; stage: number; context: boolean; }
```
Comment (vankop) [inline lib/SourceMapDevToolPlugin.js]: don't know is there any related issue

Comment (sokra) [reply lib/SourceMapDevToolPlugin.js]: This is fixed in tapable 2, which is used in the webpack next branch

Comment (vankop) [reply lib/SourceMapDevToolPlugin.js]: So your suggestion is to provide esdoc for `next` branch instead?

Comment (vankop) [reply lib/SourceMapDevToolPlugin.js]: @sokra ping

Comment (sokra) [reply lib/SourceMapDevToolPlugin.js]: Cast it to `TODO` here. This has already been fixed in the `next` branch.

``` js
tap(
  /** @type {TODO} */ ({
    name: "SourceMapDevToolPlugin",
    context: true
  }),
```

File: lib/SourceMapDevToolPlugin.js [hunk 2]
```diff
@@ -236,6 +304,9 @@ class SourceMapDevToolPlugin {
 						assetsCache.set(asset, { file, assets });
 						/** @type {string | false} */
 						let currentSourceMappingURLComment = sourceMappingURLComment;
+						/**
+						 * In css comment with `//` commenting next block as well, so workaround is to replace comment `//` with `/**\/`
```
Comment (sokra) [inline lib/SourceMapDevToolPlugin.js]: In CSS there are only `/*  */` comments, `//` comments doesn't exist.

Comment (vankop) [reply lib/SourceMapDevToolPlugin.js]: my fault, i was reading about *suggested* syntax

**Comment by sokra** [comment]
Thanks

**Pull request merged**